Require URL to match when syncing data to Merchant Center

mikkamp opened the pull request:
### Changes proposed in this Pull Request:

This PR changes the product sync behaviour and requires the URL to match to the one we used to claim the website with. This ensures that the products are only able to sync on the original site, any staging site or copy of the site will not sync product data.

If the user wishes to change the site URL then the onboarding process must be completed again to perform a new site verification/claim.

Closes #1399

### Detailed test instructions:
1. Start with a live site that's fully onboarded and synced all products to the Merchant Center
2. Create a staging site (can use a plugin like [WP Staging](https://wordpress.org/plugins/wp-staging/))
3. Update a product on the staging site
4. Check the action scheduler logs and confirm that the product was not synced
5. Check the same product in the Merchant Center (might need to wait a while before it's updated) and confirm that the URL's are still from the original site
6. Use the filter `add_filter( 'woocommerce_gla_ready_for_syncing', '__return_true' );`
7. Update a product on the staging site
8. Check the action scheduler logs and confirm that this time the product did sync
9. Check the original site and confirm that product syncing continues to function as it was when updating products

### Additional details:

In addition to the tests for the added / changed code. There was no unit test present for the MerchantCenterService class, so I added it in this PR before adding the additional tests for this class.

### Changelog entry
* Fix - Prevent product sync if the site URL does not match the originally claimed URL.

mikkamp committed:
Add option to store the hashed URL

mikkamp committed:
Helper function to retrieve the hashed URL

mikkamp committed:
Check if urls match so we are ready for syncing

mikkamp committed:
Prevent syncing when we are not sync ready

mikkamp committed:
Fix changed tests

mikkamp committed:
Tests for get_claimed_url_hash

mikkamp committed:
Unit tests for MerchantCenterService

mikkamp committed:
Tests for is_ready_for_syncing

mikkamp committed:
Confirm we clear claimed URL hash when switching

ianlin reviewed (approved):
Thanks for fixing this! I could reproduce this issue using `develop` branch, and confirmed the issue is solved using the branch of this PR. The products will only be synced to merchant centre when using the production site.

Just a note while I was testing for it:

Normally I used Ngrok to get a public domain (e.g. `ian-local-test-wp.ngrok.io`) while onboarding setup and using this public domain to claim the url. After that I used a local domain (e.g. `wp59wc61.test`) points to 127.0.0.1 for other operations since it's faster. If I use the local domain `wp59wc61.test` to update the product it will not sync to merchant centre either.

mikkamp commented:
Thanks for the review.

> Just a note while I was testing for it...
I have the same setup, which is why I added the filter `woocommerce_gla_ready_for_syncing` to override it for local testing. I've also added the filter to our [wiki](https://github.com/woocommerce/google-listings-and-ads/wiki/Testing)
